com.unity.visualeffectgraph/Editor/GraphView/VFXComponentBoard.cs

```diff
@@ -747,28 +708,25 @@ static bool IsDefaultEvent(string evt)
             return evt == VisualEffectAsset.PlayEventName || evt == VisualEffectAsset.StopEventName;
         }
 
-        IEnumerable<String> GetEventNames()
+        IEnumerable<string> GetEventNames()
         {
-            foreach (var context in controller.contexts.Select(t => t.model).OfType<VFXContext>())
-            {
-                foreach (var name in RecurseGetEventNames(context))
-                    yield return name;
-            }
+            return controller?.contexts.SelectMany(x => this.RecurseGetEventNames(x.model)) ?? Enumerable.Empty<string>();
         }
 
-        IEnumerable<String> RecurseGetEventNames(VFXContext context)
+        IEnumerable<string> RecurseGetEventNames(VFXContext context)
         {
-            if (context is VFXBasicEvent)
+            switch (context)
             {
-                if (!IsDefaultEvent(name))
-                    yield return (context as VFXBasicEvent).eventName;
-            }
-            else if (context is VFXSubgraphContext)
-            {
-                foreach (var subContext in (context as VFXSubgraphContext).subChildren.OfType<VFXContext>())
+                case VFXBasicEvent basicEvent when !IsDefaultEvent(name):
```

Comment:
Nice, I wasn't aware of this new condition : https://docs.microsoft.com/fr-fr/dotnet/csharp/language-reference/statements/selection-statements